### ISSUE TITLE
I think some bug in  tf.contrib.layers.l2_regularizer

### ISSUE THREAD

**Issue by XeMinZa** [tensorflow/tensorflow#15834]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:win10
- **TensorFlow installed from (source or binary)**:install
- **TensorFlow version (use command below)**:1.5
- **Python version**: 3.5
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

### Describe the problem
when I call the  tf.contrib.layers.l2_regularizer(0.5)(w), I was toll that "Expected int64, got 0.5 of type 'float' instead." But the doc says that it need a float clearly and as a weight it can't be an integer.

### Source code / logs
` l2 = 0
for w in tf.global_variables():
    l2 += tf.contrib.layers.l2_regularizer(0.5)(w)
loss = tf.losses.softmax_cross_entropy(onehot_labels=one_hot_labels,logits=result_vector)+l2`

**Comment by ppwwyyxx** [comment]
Your variables have int64 type.
